extensions/ros2/launch

```diff
@@ -13,6 +13,16 @@ function source_with_prefix() {
 # Save off parameters, the sourced setup scripts may manipulate them.
 original_args=("$@")
 
+if [ -d "$SNAP/opt/ros/underlay_ws/opt/ros/$ROS_DISTRO" ]; then
+  source_with_prefix "$SNAP/opt/ros/underlay_ws/opt/ros/$ROS_DISTRO"
+fi
+if [ -d "$SNAP/opt/ros/underlay_ws/opt/ros/snap" ]; then
+  source_with_prefix "$SNAP/opt/ros/underlay_ws/opt/ros/snap"
+fi
+if [[ -d "$SNAP/opt/ros/underlay_ws/opt/ros/$ROS_DISTRO" || -d "$SNAP/opt/ros/underlay_ws/opt/ros/snap" ]]; then
+  export LD_LIBRARY_PATH="$SNAP/opt/ros/underlay_ws/usr/lib/x86_64-linux-gnu:${LD_LIBRARY_PATH}"
```

Comment:
Sure. However I'm not familiar with the notation, isn't it the same as `${LD_LIBRARY_PATH-}`?

Reply:
No, with the minus it's to use a default value (in your case "") but you will still have the ":". 
For example, `${LD_LIBRARY_PATH-/bin}` will return `$LD_LIBRARY_PATH` is defined `/bin` otherwise.

while `/bin${LD_LIBRARY_PATH:+:${LD_LIBRARY_PATH}}` will append `${LD_LIBRARY_PATH}` to `/bin` only if `LD_LIBRARY_PATH` exist.

Full doc is available [here](https://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html#Shell-Parameter-Expansion)

Reply:
Gotcha :+1: